rust/hyperlane-core/src/types/message.rs

```diff
@@ -34,6 +36,34 @@ pub struct HyperlaneMessage {
     pub body: Vec<u8>,
 }
 
+impl Debug for HyperlaneMessage {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "HyperlaneMessage {{ id: {:?}, version: {}, nonce: {}, origin: {}, sender: {}, destination: {}, recipient: {}, body: 0x{} }}",
+            self.id(),
+            self.version,
+            self.nonce,
+            self.origin,
+            fmt_address_for_domain(self.origin, self.sender),
+            self.destination,
+            fmt_address_for_domain(self.destination, self.recipient),
+            hex::encode(&self.body)
+        )
+    }
+}
+
+impl Display for HyperlaneMessage {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "HyperlaneMessage {{ id: {:?}, nonce: {}, .. }}",
```

Comment:
Noting that here we use `id` vs `message_id`, in case that helps tiebreak

Reply:
Yep, thanks